### PR TITLE
Default test switch

### PR DESCRIPTION
**Pull request by jsantiago-eProsima**
## Contributor Checklist
- [ ] Commit messages follow the project guidelines. 
- [ ] The code follows the style guidelines of this project. 
- [ ] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [ ] Any new/modified methods have been properly documented using Doxygen. 
- [ ] Fast DDS test suite has been run locally. 
- [ ] Changes are ABI compatible. 
- [ ] Changes are API compatible. 
- [ ] Documentation builds and tests pass locally. 
- [ ] New feature has been added to the `versions.md` file (if applicable).
- [ ] New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by MiguelBarro — DISMISSED** [review]
LGTM
We've learned we cannot mix gcc sanitizer with clang one.

File: test/blackbox/CMakeLists.txt [hunk 1]
```diff
@@ -194,9 +194,9 @@ macro(add_blackbox_gtest)
     endif()
 endmacro()
 
-option(RTPS_API_TESTS "Enable tests using RTPS API" ON)
-option(FASTRTPS_API_TESTS "Enable tests using FastRTPS API" ON)
-option(FASTDDS_PIM_API_TESTS "Enable tests using FastDDS API" OFF)
+option(RTPS_API_TESTS "Enable tests using RTPS API" OFF)
```
Comment (MiguelCompany) [inline test/blackbox/CMakeLists.txt:197]: RTPS is part of the API and has not been deprecated, so we should not have this OFF by default, should we?

Comment (EduPonz) [reply test/blackbox/CMakeLists.txt:197]: We should definitely NOT have this OFF by default

Comment (jsantiago-eProsima) [reply test/blackbox/CMakeLists.txt]: Re-enabled these tests on commit 4cc6eb3

**Review by MiguelCompany — APPROVED** [review]
LGTM

**Pull request merged**